### PR TITLE
Bump SDL_image to 2.7.0

### PR DESCRIPTION
**Pull request by yunline**
Continue from #2182
To support animation

### CODE REVIEW

**Comment by oddbookworm** [comment]
```python
>>> pygame.print_debug_info()
Platform:               Windows-10-10.0.22621-SP0
System:                 Windows
System Version:         10.0.22621
Processor:              AMD64 Family 23 Model 8 Stepping 2, AuthenticAMD
Architecture:           Bits: 64bit     Linkage: WindowsPE
Driver:                 Display Not Initialized

Python:                 CPython
pygame version:         2.3.0.dev3
python version:         3.11.1

SDL versions:           Linked: 2.26.4  Compiled: 2.26.4
SDL Mixer versions:     Linked: 2.6.3   Compiled: 2.6.3
SDL Font versions:      Linked: 2.20.2  Compiled: 2.20.2
SDL Image versions:     Linked: 2.7.0   Compiled: 2.7.0
Freetype versions:      Linked: 2.11.1  Compiled: 2.11.1
```

**Comment by oddbookworm** [comment]
Seems to work perfectly locally

**Review by oddbookworm — APPROVED** [review]

**Comment by oddbookworm** [comment]
I'll test further on other OS's at some point today

File: .github/workflows/build-windows.yml [hunk 1]
```diff
@@ -36,8 +36,62 @@ concurrency:
...
+          # set hash, if hash changed, build again
+          key: windep-${{ hashFiles('buildconfig/windependencies/**') }}-${{ matrix.winarch }}
+
+      - name: Set up Visual Studio 2022 Build Tools
+        if: steps.windep-cache.outputs.cache-hit != 'true'
+        uses: microsoft/setup-msbuild@v1.0.2
+        with:
+          vs-version: '17'
+
+      - name: Install CMake
```
Comment (Starbuck5) [inline .github/workflows/build-windows.yml:73]: According to this page (https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md) Cmake is already installed.

File: buildconfig/Setup.SDL2.in [hunk 2]
```diff
@@ -25,7 +23,6 @@ DEBUG =
 #everything you can, but you can ignore ones you don't have
 #dependencies for, just comment them out
 
-imageext src_c/imageext.c $(SDL) $(IMAGE) $(PNG) $(JPEG) $(DEBUG)
```
Comment (Starbuck5) [inline buildconfig/Setup.SDL2.in:28]: I'd rather not split this into win/non-win versions.

Jpeg is not used directly by imageext anymore (removed by me a while back), it's just vestigial in buildconfig. So it should work to only link libpng both places.

**Review by MyreMylar — CHANGES_REQUESTED** [review]
Looks like this needs a bit of love to keep up with main, and some adjustments from @Starbuck5's feedback. I think @ankith26 should also look it over as I know they are doing stuff with windows dependencies at the moment.

**Comment by MyreMylar** [comment]
Seems like this is superseded by moving to version 2.8.0 in this PR: https://github.com/pygame-community/pygame-ce/pull/2596

**Review by ankith26 — COMMENTED** [review]
We will be moving away from the old python buildconfig, and on windows we still want to keep deps precompiled and not compiled at any pygame dev machines.

This PR did some nice work, but we are on SDL_image 2.8.0 on main, doing things differently than this PR. Thanks for the work anyways.

**Comment by Starbuck5** [comment]
Yes, thank you yunline for working on tackling this important item. I just wasn't sure what the right strategy was.

**Pull request closed without merging**